File: solutions/_724_find_pivot_index/solution.java
package _724_find_pivot_index;

class Solution {

  public int pivotIndex(int[] nums) {
    int sum = 0;
    for (int num : nums) {
      sum += num;
    }

    int leftSum = 0;
    for (int i = 0; i < nums.length; i++) {
      int rightSum = sum - leftSum - nums[i];
      if (leftSum == rightSum) {
        return i;
      }
      leftSum += nums[i];
    }

    return -1;
  }
}
/*
https://leetcode.com/problems/find-pivot-index/
*/

File: solutions/_724_find_pivot_index/solution.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total = sum(nums)

        leftSum = 0
        for i in range(len(nums)):
            rightSum = total - nums[i] - leftSum
            if leftSum == rightSum:
                return i
            leftSum += nums[i]
        return -1


# https://leetcode.com/problems/find-pivot-index/
